### PR TITLE
Feat/check allownace on create

### PR DESCRIPTION
**Pull request by Julio-SDSharp**
# Enhanced Allowance Management

## Overview

This pull request delivers a more robust and user-friendly allowance management experience, incorporating the following key improvements:

**Key Enhancements:**

- **Streamlined Allowance Deletion:**
    - Awaits completion before closing to ensure data integrity.
    - Displays clear error messages to guide users in case of failures.
- **Optimized Storage:**
    - Removes unnecessary fields to reduce storage footprint and enhance efficiency.
- **Enhanced Testing and Verification:**
    - Adds a "Check Allowance" button for quick verification within the add allowance process.
    - Automatically adds verified allowances to the table for seamless integration.
    - Prevents duplication of allowances, ensuring consistency in the data.
    - Enables modification of existing allowances directly from the test allowance screen.

**Completed Tasks:**

- [x] Remove identified unnecessary fields from storage.
- [x] Implement awaiting mechanism before closing allowance deletion.
- [x] Integrate clear error messaging for deletion failures.
- [x] Integrated "Check Allowance" button within add allowance process.
- [x] Implemented automatic table addition for verified allowances.
- [x] Added safeguards against allowance duplication.
- [x] Enabled modification of existing allowances during testing.

## Next Steps

- Complete pending tasks to finalize allowance deletion and storage optimization.
- Conduct thorough testing to ensure all enhancements function as intended.
- Request code review for merge

### CODE REVIEW

File: frontend/pages/home/components/ICRC/allowance/CreateForm/index.tsx [hunk 1]
```diff
@@ -23,50 +37,118 @@ export default function CreateForm() {
...
     </form>
   );
 
+  async function onCheckAllowance(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    try {
+      setLoading(true);
+      event.preventDefault();
+      setFullAllowanceErrorsAction([]);
+
+      if (!allowance.asset.address || !allowance.asset.decimal)
```
Comment (OliverSDS) [inline frontend/pages/home/components/ICRC/allowance/CreateForm/index.tsx:94]: can we add comments? validation is complex and could cause issues if not clear

Comment (OliverSDSharp) [inline frontend/pages/home/components/ICRC/allowance/CreateForm/index.tsx:95]: can we add comments? validation is complex and could cause issues if not clear

File: frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx [hunk 2]
```diff
@@ -50,27 +52,17 @@ export default function TransactionAmount() {
...
-
-      if (!isValid) {
-        setErrorAction(TransactionValidationErrorsEnum.Values["error.invalid.amount"]);
-        return;
-      }
-      setAmountAction(amount);
+  function onChangeAmount(e: ChangeEvent<HTMLInputElement>) {
+    const amount = e.target.value.trim();
+    setAmountAction(amount);
+    const isValidAmount = validateAmount(amount, Number(sender.asset.decimal));
```
Comment (OliverSDSharp) [inline frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx:58]: is value is "" will get NaN. please verify

File: frontend/pages/home/hooks/useDeleteAllowance.ts [hunk 3]
```diff
@@ -1,55 +1,64 @@
...
-      return { success: true };
-    } catch (error) {
-      console.error(error);
-      return { success: false, error };
-    } finally {
+      setIsPending(true);
+      const subAccount = asset.subAccounts.find(
+        (subAccount) => subAccount.sub_account_id === selectedAllowance.subAccountId,
+      );
+      if (!subAccount) throw "no sub account to delete";
```
Comment (OliverSDSharp) [inline frontend/pages/home/hooks/useDeleteAllowance.ts]: can it be translated? those it show on a dialog or something?

**Review by OliverSDSharp — CHANGES_REQUESTED** [review]

File: frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx (hunk 2 above)
Comment (0riion) [reply frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx:58]: Verified, if "" the isValidAmount will be still boolean.

File: frontend/pages/home/components/ICRC/allowance/CreateForm/index.tsx (hunk 1 above)
Comment (Julio-SDSharp) [reply frontend/pages/home/components/ICRC/allowance/CreateForm/index.tsx:95]: added

File: frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx (hunk 2 above)
Comment (Julio-SDSharp) [reply frontend/pages/home/components/ICRC/detail/transaction/SendForm/TransactionAmount.tsx:58]: Verified, if amount is "" will get still boolean.

File: frontend/pages/home/hooks/useDeleteAllowance.ts (hunk 3 above)
Comment (Julio-SDSharp) [reply frontend/pages/home/hooks/useDeleteAllowance.ts]: It was used for test purpose, not for user feedback. Removed.

**Review by OliverSDSharp — APPROVED** [review]

**Pull request merged**